feat: add user creation endpoint

HijoDeHonor opened the pull request:
Implemented POST endpoint '/users' for user creation. Added corresponding Express routes and unit tests to verify functionality.

HijoDeHonor committed:
feat: add user creation endpoint

Implemented POST endpoint '/users' for user creation. Added corresponding Express routes and unit tests to verify functionality.

PiewDev commented on server-with-mock.js:
Is this duplicate file? is equal to that of sql

HijoDeHonor replied:
no, this import the UserModel from another file to change if uses a MySQL o local files

PiewDev commented on test/user/userControler.test.js:
don't use Magic Strings, uses constants

PiewDev commented on test/user/userControler.test.js:
Separate the test with comments:

Arrange
Act
Assert

PiewDev commented on test/user/validationUser.test.js:
don't use Magic Strings, uses constants.

Separate the test with comments:

Arrange
Act
Assert

PiewDev commented on users/mockModel.js:
Is this file necesary?

PiewDev commented on users/mySqlModel.js:
```diff
@@ -0,0 +1,39 @@
+  static async create ({ input }) {
+    const { userName, password } = input
+    try {
+      await connection.query(
```
Needs an extra check on whether the user exists

PiewDev replied:
Needs an extra check on whether the user exists

PiewDev commented on users/mySqlModel.js:
```diff
@@ -0,0 +1,39 @@
+import mysql from 'mysql2/promise'
+import { ERROR_CREATING_USER } from '../utils/textConstants'
+
+const DEFAULT_CONFIG = {
```
Remove these configurations to a separate file

PiewDev replied:
Remove these configurations to a separate file

PiewDev commented on users/registerUser/userControler.js:
```diff
@@ -0,0 +1,37 @@
+    const result = validateUser(userName, password);
+    if (!result.success) {
+      throw new InvalidDataError(INVALID_DATA, USERS);
+    }
```
remove this try-catch should be handled by ErrorHandler

PiewDev replied:
remove this try-catch should be handled by ErrorHandler

PiewDev commented on users/registerUser/userControler.js:
```diff
@@ -0,0 +1,37 @@
+    }
+  }
+}
+
```
Remove these class to a separate file

PiewDev replied:
Remove these class to a separate file

PiewDev commented on users/userSchema.js:
```diff
@@ -0,0 +1,24 @@
+
+
+
+
```
Remove this comments

PiewDev commented on middlewares/errorHandler/errorhandler.js:
```diff
@@ -1,5 +1,12 @@
   const message = statusCode >= 500 ? 'Internal Server Error' : error.message
   return res.status(statusCode).send(message)
 }
+
```
Are you importing this Constants?

PiewDev commented on server-with-mock.js:
Is this duplicate file? is equal to that of sql

PiewDev commented on test/user/userControler.test.js:
don't use Magic Strings, uses constants

Separate the test with comments:

Arrange
Act
Assert

PiewDev commented on test/user/validationUser.test.js:
Separate the test with comments:

Arrange
Act
Assert

PiewDev commented on users/mockModel.js:
Is this file necesary?

PiewDev reviewed (changes requested):
It's pretty good, there are only small things to change.

HijoDeHonor committed:
refactor: endpoint user creation

refactor unit test on userController, validationUser and userModel.

HijoDeHonor committed:
feat: add user creation endpoint with unit tests

Implemented POST endpoint '/api/users' for user creation. Added corresponding Express routes and unit tests to verify functionality and ensure robust testing coverage.

PiewDev commented on package-lock.json:
package.lock.json should not be saved to the repository

PiewDev commented on package.json:
```diff
@@ -16,11 +18,16 @@
-    "uuid": "10.0.0"
+    "fs": "^0.0.1-security",
+    "mysql2": "^3.10.1",
+    "sinon": "^18.0.0",
```
In the end you didn't use Sinon

PiewDev commented on test/user/userControler.test.js:
```diff
@@ -0,0 +1,85 @@
+
+    // Assert
+    expect(res.status).toBe(201);
+    expect(res.body).toEqual(uniqueUserName);
```
The response from the endpoint should be body.userName. make the change in the controller

PiewDev commented on .vscode/settings.json:
settings.json should not be saved to the repository

PiewDev reviewed (changes requested):
Very good, we need to make some small changes related to the health of the repository.

HijoDeHonor committed:
chore: update gitignore and cleanup dependencies, refactor user endpoint

- Added package-lock.json to .gitignore
- Changed the return of the user creation endpoint
- Updated corresponding unit tests
- Removed unused dependencies

PiewDev commented on .vscode/settings.json:
Add to gitignore

PiewDev commented on users/registerUser/userControler.js:
```diff
@@ -0,0 +1,21 @@
+    }
+    const newUser = await this.userModel.create({ input: result.data });
+    if (newUser) {
+      res.status(201).json({ userName: newUser });
```
My bad, the one that should return this format is the model.

PiewDev replied:
newUser should be an object with the property userName

PiewDev reviewed (changes requested):
A change that you missed from the previous one and a mistake of mine, we are even.

HijoDeHonor committed:
chore: update gitignore

HijoDeHonor committed:
refactor: update UserModel return values and corresponding tests

- Changed the return values of the UserModel
- Updated the user creation endpoint to reflect these changes
- Modified unit tests to match the updated UserModel return values
- Added package-lock.json and settings of vscode to .gitignore

PiewDev reviewed (approved):
Good job! All the changes look good
